### DISCUSSION TITLE
Authorization and multiple users

### DISCUSSION THREAD

**Discussion by kfrolov-rbk** [aminalaee/sqladmin#159] [Ideas]
Is it possible to create multiple users and give them different roles and access level?
Like manager roles, who with-read only rights in certain models. 
I was looking for something similar to Django admin but for FastAPI but not sure if SQLAdmin would be suitable because of lack of authorization functionality. 
But if you add it, this would be dramatically improve FastAPI experience for many developers

**Comment by aminalaee** [comment]
Hey thank you for the suggestion.

Right now as might know you can implement the functionality on your own, as the SQLAdmin only provides the admin functionality.
But I think having a separate package for it would be better, because there could be lots of ways to handle this and multiple tools for it. For example one would use Redis for the Auth but someone else would use the same databse, and also the same applies for the functionalities.

Do you have any suggestions?

**Reply by aryadovoy** [reply]
I'm trying to combine [fastapi-users](https://github.com/fastapi-users/) and sqladmin, but have some difficulties. Could you help with that?

**Reply by aminalaee** [reply]
Hey, yeah sure. What is the issue?

**Reply by aryadovoy** [reply]
That's how I can check in fastapi-users that user is authenticated, but I don't know how to do it with sqladmin.

```
@app.get("/authenticated-route")
async def authenticated_route(user: User = Depends(current_active_user)):
    return {"message": f"Hello {user.email}!"}
```
